pageserver/src/relish_storage/rust_s3.rs

```diff
@@ -0,0 +1,195 @@
+use s3::{bucket::Bucket, creds::Credentials, region::Region};
+use zenith_utils::zid::{ZTenantId, ZTimelineId};
+
+use crate::S3Config;
+
+use super::{RelishStorage, RelishStoragePath, S3Error};
+
+lazy_static::lazy_static! {
+    static ref UPLOAD_RUNTIME: tokio::runtime::Runtime = tokio::runtime::Builder::new_current_thread()
```

Comment:
We've had a conversation with @hlinnaka where he shared his preferences for sync interfaces for the time being, that's why I've made the new trait sync and hid the async behind this way.

It's in the lazy_static, because I wrap the whole upload loop in `catch_unwind` that does not like `Runtime` as one of the `RustS3` fields in this case.

But what's more interesting to note, due to the whole nature of the upload loop, it feels like we could actually go with totally async approach right away, including the local fs manipulations.

This way, the only sync interface would be the `schedule_upload` function that does zero async by itself anyway.
Should we go this way instead?

Reply:
I'm pretty wary of async code in general, our experience is that mixing sync and async code quickly becomes a mess, and async code is difficult to debug. That said, I agree that async would seem like a natural fit here. As long as `schedule_upload' is synchronous, I guess that isolates this enough from the rest of the codebase.